src/main/java/org/folio/dew/service/BulkEditParseService.java

```diff
@@ -0,0 +1,184 @@
+      .flatMap(proxyForCollection -> proxyForCollection.getProxiesFor().stream())
+      .map(ProxyFor::getId)
+      .collect(Collectors.toList());
+  }
+
+  private Personal getUserPersonalInfo(UserFormat userFormat) {
+    Personal personal = new Personal();
+    personal.setLastName(userFormat.getLastName());
+    personal.setFirstName(userFormat.getFirstName());
+    personal.setMiddleName(userFormat.getMiddleName());
+    personal.setPreferredFirstName(userFormat.getPreferredFirstName());
+    personal.setEmail(userFormat.getEmail());
+    personal.setPhone(userFormat.getPhone());
+    personal.setMobilePhone(userFormat.getMobilePhone());
+    personal.setDateOfBirth(getDate(userFormat.getDateOfBirth()));
+    personal.setAddresses(getUserAddresses(userFormat));
+    personal.setPreferredContactTypeId(userFormat.getPreferredContactTypeId());
+    return personal;
+  }
+
+  private Date getDate(String date) {
+    LocalDateTime localDateTime = LocalDateTime.parse(date, DateTimeFormatter.ofPattern(DATE_TIME_PATTERN));
+    return Date.from(localDateTime.atZone(UTC).toInstant());
+  }
+
+  private List<Address> getUserAddresses(UserFormat userFormat) {
+    String[] addresses = userFormat.getAddresses().split(ITEM_DELIMITER_PATTERN);
+    return Arrays.stream(addresses)
+      .parallel()
+      .map(this::getAddressFromString)
+      .collect(Collectors.toList());
+  }
+
+  private Address getAddressFromString(String stringAddress) {
+    Address address = new Address();
+    List<String> addressFields = Arrays.asList(stringAddress.split(ARRAY_DELIMITER));
+    address.setId(addressFields.get(ADDRESS_ID));
+    address.setCountryId(addressFields.get(ADDRESS_COUNTRY_ID));
+    address.setAddressLine1(addressFields.get(ADDRESS_LINE_1));
+    address.setAddressLine2(addressFields.get(ADDRESS_LINE_2));
+    address.setCity(addressFields.get(ADDRESS_CITY));
+    address.setRegion(addressFields.get(ADDRESS_REGION));
+    address.setPostalCode(addressFields.get(ADDRESS_POSTAL_CODE));
+    address.setPrimaryAddress(Boolean.valueOf(addressFields.get(ADDRESS_PRIMARY_ADDRESS)));
+    //avoid IndexOutOfBoundsException if address type id wasn't set
+    if (addressFields.size() == 9) {
+      String id = userReferenceService.getAddressTypeByDesc(addressFields.get(ADDRESS_TYPE)).getAddressTypes().iterator().next().getId();
+      address.setAddressTypeId(id);
+    }
+    return address;
+  }
+
+  private Tags getTags(UserFormat userFormat) {
+    if (isNotEmpty(userFormat.getTags())) {
+      Tags tags = new Tags();
+      List<String> tagList = Arrays.asList(userFormat.getTags().split(ARRAY_DELIMITER));
+      return tags.tagList(tagList);
+    }
+    return null;
+  }
+
+  private Map<String, Object> getCustomFields(UserFormat userFormat) {
+    if (isNotEmpty(userFormat.getCustomFields())) {
+      Map<String, Object> customFields = new HashMap<>();
+      String[] customFieldsArray = userFormat.getCustomFields().split(ITEM_DELIMITER_PATTERN);
```

Comment:
Empty strings should be excluded from processing

Reply:
In this place, we don't need to escape empty strings.